edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/AgentModule.cs

```diff
@@ -134,9 +140,22 @@ protected override void Load(ContainerBuilder builder)
                 .As<IRocksDbOptionsProvider>()
                 .SingleInstance();
 
-            // IDbStore
+            if (!this.usePersistentStorage && this.useBackupAndRestore)
+            {
+                // Backup and restore serialization
+                builder.Register(c => new ProtoBufBackupRestore())
+                    .As<IBackupRestore>()
+                    .SingleInstance();
+
+                // Backup and restore of IDbStore.
+                builder.Register(c => new DbStoreBackupRestore(c.Resolve<IBackupRestore>()))
+                    .As<IDbStoreBackupRestore>()
+                    .SingleInstance();
+            }
+
```

Comment:
Does it cause any problem if these get registered with feature off? (don't change the code if not, just asking)

Reply:
Nope, it won't cause any issues. Just that an instance will be created and never be used.